### ISSUE TITLE
[8.0] Current Currency Rate

### ISSUE THREAD

**Issue by rayer91** [odoo/odoo#12648]
Impacted versions: 8.0

Steps to reproduce: 
Add new Currency Rate  with date and exact time.

Current behavior:
The current currency rate is not a current currency rate. Because in the SQL query the parameter is the current date and not a Datetime and show the yesterday's currency rate and not today's currency rate .

Expected behavior:
Show today currency rate.

Fixing:

Replace or override in the res.currency model:
```
def _get_current_rate(self, cr, uid, ids, raise_on_no_rate=True, context=None):
...
...
...
date = context.get('date') or time.strftime('%Y-%m-%d')
```
To:
```
def _get_current_rate(self, cr, uid, ids, raise_on_no_rate=True, context=None):
...
...
...
date = context.get('datetime') or time.strftime('%Y-%m-%d %H:%M:%S')
```

**Comment by spadae22** [comment]
I can confirm this issue.

**Comment by Yenthe666** [comment]
@mart-e looks like a real issue. Can you look at this?

**Comment by TheCloneMaster** [comment]
This is because of time zone differences.  In our case we overwrite the
currency table and set the field to date, avoiding time differences, but if
you need several rates per day that is not feasible,and you will need to
adjust for time zone differences...

**Comment by mart-e** [comment]
Hello,

This was already fixed in 9.0 at 55fa0912 and I have backported it to 8.0 at 2c19e9be9691ff02e80ddca8cda05ba87fd16550
Thanks
